src/App.js

```diff
@@ -1,5 +1,136 @@
+const { Console } = require('@woowacourse/mission-utils');
+const { getRandomNumbers, convertStringsToNumbers } = require('./utils');
+const {
+  HINT_MESSAGE_INVALID_PARAM,
+  INPUT_FAIL_ERROR_MESSAGE,
+} = require('./errorMessage');
+
+const BASEBALL_NUMBER_CNT = 3;
+
 class App {
-  play() {}
+  constructor() {
+    this.computer = {
+      randomNumbers: [],
+    };
+  }
+
+  play() {
+    this.startGame();
+  }
+
+  startGame() {
+    this.print('숫자 야구 게임을 시작합니다.');
+
+    this.setRandomNumbers();
+    this.numberPrediction();
+  }
+
+  numberPrediction() {
+    this.readLine('숫자를 입력해주세요 : ', (answer) => {
+      const answers = answer.trim().split('');
+      const rightAnswers = this.computer.randomNumbers;
+
+      if (answers.length === BASEBALL_NUMBER_CNT) {
+        const numbers = convertStringsToNumbers(answers);
+
+        const { strike, ball } = this.getResult(numbers, rightAnswers);
+        const hint = this.getHintMessage(strike, ball);
+
+        this.print(hint);
+
+        this.checkEndGame(strike);
+      } else {
+        throw new Error(INPUT_FAIL_ERROR_MESSAGE);
+      }
+    });
+  }
+
+  checkEndGame(strike) {
+    if (strike === BASEBALL_NUMBER_CNT) {
+      this.endGameInputMessage();
+    } else {
+      this.numberPrediction();
+    }
+  }
+
+  endGame(answer) {
+    const trimAnswer = answer.trim();
+
+    if (trimAnswer === '1') {
+      this.startGame();
+    } else if (trimAnswer === '2') {
+      Console.close();
+    } else {
+      throw new Error(INPUT_FAIL_ERROR_MESSAGE);
+    }
+  }
```

Comment:
```suggestion
  endGame(answer) {
    const trimAnswer = answer.trim();
    if (trimAnswer !== '1' && trimAnswer !== '2') {
      throw new Error(INPUT_FAIL_ERROR_MESSAGE);
    }
    (trimAnswer === '1') ? this.startGame() : Console.close();
  }
```

이런식으로 먼저 예외값들을 처리해준다면 else를 없앨 수 있을것같습니다 !!